pkgs/build-support/substitute/substitute.nix

```diff
@@ -6,7 +6,10 @@ args:
 # The `replacements` attribute should be a list of list of arguments
 # to `substitute`, such as `[ "--replace" "sourcetext" "replacementtext" ]`
 stdenvNoCC.mkDerivation ({
-  name = if args ? name then args.name else baseNameOf (toString args.src);
+  # unsafeDiscardStringContext is required here, because calling `lib.getName`
+  # on the resulting derivation gives the following error:
+  # "the string '50-coredump.conf' is not allowed to refer to a store path (such as '!out!/nix/store/zgf4bx2jrw6ysjlr3qf4i942z64d6559-systemd-253.5.drv')"
+  name = if args ? name then args.name else (builtins.unsafeDiscardStringContext (builtins.baseNameOf (toString args.src)));
```

Comment:
Why is this suddenly required? Why is that error thrown? The comment doesn't really explain that

Reply:
I honestly don't know more about it other than that error message. I think the issue is always there, but until now nobody was calling `getName` on all our packages so it never surfaced.

Reply:
Turns out I can work around this by wrapping the values in `toString`, then passing in `null` will simply print null instead of throwing a type error.